Feature/create project

navjot-nangia opened the pull request:
addresses #94

navjot-nangia committed:
added luminance parameter to work on missing_white

navjot-nangia committed:
corrected method name in missing_white

mkutu committed:
Merge pull request #92 from precision-sustainable-ag/missing_white_refine

added luminance parameter to work on missing_white

mkutu committed:
skeleton structure for creating project by filtering by species

navjot-nangia committed:
Refactor project creation and filtering by species; add new configuration files and update paths

navjot-nangia committed:
Merge branch 'develop' into feature/create_project

navjot-nangia commented:
@mkutu 
Either the way you created or the way i merged this branch `Feature/create_project` is weird. Looks like it is being developed on `Feature/relabel` and combining all commits in one branch. We can talk about it tomorrow.

mkutu committed:
cleaned up entry point

mkutu committed:
simplified project creation and image copying using new db column

mkutu committed:
simplified sampling and clean connecting and closing of db

mkutu committed:
removed unused  paths in config

mkutu commented on src/create_projects_utils/filter_by_species.py:
```diff
@@ -0,0 +1,93 @@
+            developed_image_path = row["developed_image_path"]
+            source_path = self.lts_source_dir / developed_image_path
+            if source_path.exists():
+                log.info(f"Copying {developed_image_path.name} from {source_path} to {dst_dir}")
```
shoudl be 
```python
log.info(f"Copying {source_path.name} from {source_path} to {dst_dir}")
```

mkutu commented on src/create_projects_utils/filter_by_species.py:
```diff
@@ -0,0 +1,93 @@
+
+    def _load_db(self) -> None:
+        df = pd.read_sql_query("SELECT * FROM field_data", self.conn)
+        filtered_df = df[df['extension'] == 'jpg']
```
should be
```python
        filtered_df = df[(df['extension'] == 'jpg') & (df['is_preprocessed'])]
```

mkutu committed:
fix: correct logging message and enhance image filtering in database query

mkutu committed:
filter out without mat and color checker images from sample

mkutu committed:
Skeleton filter_by_species.py changes. Non-workling changes. filter_by_species.py needs updating.

mkutu commented:
ongoing tasks include:
1. update filter_by_species.py by creating the csv using the sampled_df.
  2. temp_db.csv should be the `sampled_df` with the only thing updated is the `image_path` which would now be in the local project directory. 
6. cleaning up which includes removing unecessary code, add documentation
7. improve logging

navjot-nangia committed:
- updated the DataFrame with the local path using the image_id index
- updated documentation
- updated logging

navjot-nangia committed:
removed donwnscaled images as we are working with full scale images now

navjot-nangia committed:
- minor doc changes
- removed cash and covercrops from the yaml file to track species
- updated yaml file for species tracking

navjot-nangia committed:
- minor

navjot-nangia committed:
minor

navjot-nangia commented:
@mkutu 
This is ready for your review. 
I have added `local_image_path` column in temp_csv. Give it a try.

mkutu committed:
changed column path to local_developed_image_path

mkutu committed:
minor fix: simplify DataFrame handling and improve image filtering logic
